### DISCUSSION TITLE
How to calculate free space correctly when some torrents haven't been preallocated

### DISCUSSION THREAD

**Discussion by wuxunfeng666** [transmission/transmission#2007] [Q&A] [answered]
I wrote a python script to download some torrents from PT website and add them to transmission. It will check is there enough free space left for each torrent adding and delete some old torrents if not.

The question is, if the previous torrent dosn't download, or downloads a little bit only, which exists in the memory cache, the free space returned by transmission rpc won't change. So the script will download and add ALL torrents which size is less than the free space calculated in the beginning. And many of them will fail due to 'no enough space left' later.

Is there a way to calculate free space correctly? Or can transmission preallocate space after adding torrent immediately?

I'm using transmission daemon 2.94 on debian buster and 3.00 on windows 10.

Thanks.

**Comment by ile6695** [comment]
Transmission-daemon has config file in `/etc/transmission-daemon/settings.json` that has setting for preallocation.
You could maybe use `transmission-show <.torrent file>` to get the size of file if you need it.

**Reply by wuxunfeng666** [reply]
I've tried different preallocation values (1 - sparse, 2 - full), but transmission always preallocate space after the memory cache is full and gonna write them to disk. before that, free space calculated by transmission won't change.

I can get the size of file from .torrent. that's ok, not a problem.

My script might work like this:
``` python3
bt_cli = transmission_rpc.Client()

for i in bt_list_downloaded_from_pt_website():

    # in my case, this is almost the same value 'cause the adding is so fast. I think maybe
    # none of them can find the peers, then download enough data to full fill the memory cache,
    # then transmission can preallocate space, finally write the cache into file in such a short time?
    free_space = bt_cli.free_space(DOWNLOAD_DIR)

    if i.size > free_space and not delete_oldest_bt(need_space=i.size - free_space):
        continue  # too large, give up this

    bt_cli.add_torrent(i)
    # I wish transmission can preallocate space here, so free space calculated next time can change.
```

**Comment by ckerr** [comment, marked as answer]
There's not currently a way to do this inside Transmission itself. It would need to be added externally e.g. by your script.
